sklearn/model_selection/_validation.py

```diff
@@ -231,12 +235,16 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
             estimator.fit(X_train, **fit_params)
         else:
             estimator.fit(X_train, y_train, **fit_params)
+        fit_time = time.time() - start_time
 
     except Exception as e:
+        # Note fit time as time until error
+        fit_time = time.time() - start_time
         if error_score == 'raise':
             raise
         elif isinstance(error_score, numbers.Number):
             test_score = error_score
+            score_time = time.time() - fit_time
```

Comment:
I don't think score_time is negative here, but I do think this would be clearer, given the error cases, if you just calculated overall time and `score_time` and subtracted to get `fit_time`.